File: pinyougou-cart-interface/src/main/java/com/pinyougou/cart/service/CartService.java
package com.pinyougou.cart.service;

import java.util.List;

import com.pinyougou.pojogroup.Cart;

public interface CartService {
	
	/**将商品添加到购物车
	 * @param list 购物车对象列表
	 * @param itemId 商品id（SKU ID）
	 * @param num 商品数量
	 * @return
	 */
	public List<Cart> addGoodsToCartList(List<Cart> list, Long itemId, Integer num);
	
	/**从redis中查询购物车
	 * @param username 用户名
	 * @return 如果有购物车则返回对应购物车列表，如果没有则返回空list
	 */
	public List<Cart> findCartListFromRedis(String username);
	
	/**将购物车保存到redis
	 * @param username 用户名
	 * @param cartList 购物车列表
	 */
	public void saveCartListToRedis(String username,List<Cart> cartList);
	
	/**
	 * 合并购物车
	 * @param cartList1
	 * @param cartList2
	 * @return 返回合并后的购物车
	 */
	public List<Cart> mergeCartList(List<Cart> cartList1,List<Cart> cartList2);

}

File: pinyougou-cart-web/src/main/webapp/js/service/cartService.js
//购物车服务层
app.service('cartService',function($http){
	//查询购物车列表
	this.findCartList=function(){
		return $http.get('cart/findCartList.do');		
	}
	
	//添加商品到购物车
	this.addGoodsToCartList=function(itemId,num){
		return $http.get('cart/addGoodsToCartList.do?itemId='+itemId+'&num='+num);
	}
	
	//求合计
	this.sum=function(cartList){//cartList 购物车列表		
		var totalValue={totalNum:0, totalMoney:0.00 };//合计实体
		for(var i=0;i<cartList.length;i++){//循环商家购物车
			var cart=cartList[i];
			for(var j=0;j<cart.orderItemList.length;j++){//循环该商家所有商品
				var orderItem=cart.orderItemList[j];//购物车明细
				totalValue.totalNum+=orderItem.num;
				totalValue.totalMoney+= orderItem.totalFee;
			}				
		}
		return totalValue;
	}

});
